core/node/api_server/src/tx_sender/proxy.rs

```diff
@@ -245,11 +280,7 @@ impl TxSink for TxProxy {
         // But before we do that, save the tx to cache in case someone will request it
         // Before it reaches the main node.
         self.save_tx(tx.clone()).await;
-        self.submit_tx_impl(tx).await?;
-        // Now, after we are sure that the tx is on the main node, remove it from cache
-        // since we don't want to store txs that might have been replaced or otherwise removed
-        // from the mempool.
-        self.forget_tx(tx.hash()).await;
+        self.submit_tx_impl(tx).await?; // FIXME: shouldn't tx be removed from cache on error?
```

Comment:
Subj. Unless I'm missing something, if submitting a transaction fails, currently (including in `main`) the transaction will remain in `TxCache` indefinitely.

Reply:
I guess it should

Reply:
It could make sense to retain it locally if there's some logic pushing such txs to the main node asynchronously and the original EN RPC call doesn't return an error. Neither of these 2 conditions hold (in the `main` branch as well); thus, it looks weird to retain the transaction in the cache even after returning an error to the client and having no way to remove this transaction from the cache later (esp. in `main`; with this PR, it could get removed eventually if it gets to the main node and is included into a block, or if we implement some sort of expiration / LRU logic for the cache).

Reply:
Sorry, I was referring to the comment in the code -- e.g. I think that tx should be removed from cache on error.

Reply:
Got it. I'm just thinking aloud about why it would make sense to remove, and the corresponding pros / cons.